pose detection works

NotCras opened the pull request:
Also, major edits to the package's structure - more in line with conventional standard now. Making it so that its able to be installed on pypi.

NotCras committed:
improved readme

NotCras committed:
laying groundwork for pose detection functionality

NotCras committed:
pose detect on a single set of corners works

NotCras committed:
working on posedetect for a stream of data, however running into bugs

consolidated code into aruco_func file for straightforward aruco code analysis functions in terminal

NotCras committed:
pose detection works on a stream

outputs pose as a relative position, based on first valid position

NotCras committed:
fleshing out ArucoLoc obj

NotCras committed:
cleaning up a little before continuing

NotCras committed:
moved things around for pypi

NotCras committed:
more cleaning, reorganizing files to be more correct

NotCras committed:
Merge branch 'poses' into main

NotCras committed:
updated version

NotCras committed:
added multiple id analysis to aruco_func

also cleaned up some files

NotCras committed:
added documentation using pdoc3

NotCras committed:
fixed typo

NotCras committed:
edit to version num

JCampbell9 reviewed (approved):
I had to manually install openvc, pandas, and matplotlib prior to install this package with pip, should be added in the readme.

ac.folder_loc doesn't work but ac.file_loc change variable name.

corner_analysis() appears to be functioning properly.

how to use the pose data is not in the readme and was not tested.

NotCras committed:
updated documentation, setup.py

also matplotlib added as dependency, moved functions from ArucoFunc to ArucoHelper because it made more sense. Other cleanup like variables names

NotCras committed:
editing download url and used wrong python version 3.9->3.8

NotCras commented:
This is due to an issue with installing opencv. For now I will make a note of that in the readme.
Changed variable name.
improved readme, now has pose data process

NotCras committed:
missing arucohelper in init.py

NotCras committed:
incrementing version

uploading fix to pypi so arucohelper is available

NotCras committed:
one more version spot

NotCras committed:
fixed typo

works correctly now

NotCras committed:
fixed typo

NotCras committed:
packaging everything for aruco_helper fix

NotCras committed:
added show_image and load_poses directly

NotCras committed:
edit to readme

NotCras committed:
fixed aruco_func single analysis

NotCras committed:
added functions to aruco helper

changed init.py for directly importing functions

NotCras committed:
improved readme - quick start guide

NotCras committed:
removed self reference from load_poses and load_corners

NotCras committed:
catching up on adequate testing for poses

NotCras committed:
added pathlib to tests, fixed minor errors

NotCras committed:
update to documentation

NotCras committed:
fix to setup.py file

NotCras commented:
made some more changes after offline discussion. Thoughts?
